core/src/main/com/github/mmorihiro/core/MakeNewGeneration.kt

```diff
@@ -3,6 +3,48 @@ package com.github.mmorihiro.core
 
 internal class MakeNewGeneration {
     fun get(array: Array<BooleanArray>): Array<BooleanArray> {
-        return array
+        val copy = copy(array)
+        for (i in array.indices) {
+            for (j in array[0].indices) {
+                val pair = Pair(array[i][j], countSurroundings(array, i, j))
+
+                when (pair) {
+                    Pair(false, 3) -> copy[i][j] = true
+                    Pair(true, 2), Pair(true, 3) -> copy[i][j] = true
+                    Pair(true, 0), Pair(true, 1) -> copy[i][j] = false
+                    else ->
+                        if (pair.first == true && pair.second >= 4)
```

Comment:
I use the alternative way that uses `if` statement although this code is in `when` statement.
I don't know how to implement it without using `if`.